### PR TITLE
steam-play-none: init at 0-unstable-2022-12-15

### PR DESCRIPTION
**Pull request by MatthewCroughan**
## Description of changes

For using things like steamvr without them crashing in unexpected ways

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Atemu — APPROVED** [review]
Diff LGTM and awfully familiar ;p

Waiting for input from @Scrumplex as they're the upstream maintainer :)

File: pkgs/by-name/st/steam-play-none/package.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  meta = {
+    description = ''
+      Steam Play Compatibility Tool to run games as-is
+
+      (This is intended for use in the `programs.steam.extraCompatPackages` option only.)
+    '';
+    homepage = "https://github.com/Scrumplex/Steam-Play-None";
+    license = lib.licenses.cc0;
+    maintainers = with lib.maintainers; [ matthewcroughan scrumplex ];
```
Comment (Scrumplex) [inline pkgs/by-name/st/steam-play-none/package.nix:35]:
```suggestion
    maintainers = with lib.maintainers; [ matthewcroughan Scrumplex ];
```

File: pkgs/by-name/st/steam-play-none/package.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    platforms = [ "x86_64-linux" ];
```
Comment (Scrumplex) [inline pkgs/by-name/st/steam-play-none/package.nix:36]: Technically this runs anywhere where bash exists ^^

File: pkgs/by-name/st/steam-play-none/package.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    sourceProvenance = [ lib.sourceTypes.binaryNativeCode ];
```
Comment (Scrumplex) [inline pkgs/by-name/st/steam-play-none/package.nix:37]: Technically not a binary ^^

File: pkgs/by-name/st/steam-play-none/package.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    runHook preBuild
+
+    # Make it impossible to add to an environment. You should use the appropriate NixOS option.
+    # Also leave some breadcrumbs in the file.
+    echo "${finalAttrs.name} should not be installed into environments. Please use programs.steam.extraCompatPackages instead." > $out
+
+    ln -s $src $steamcompattool
+
+    runHook postBuild
+  '';
```
Comment (Scrumplex) [inline pkgs/by-name/st/steam-play-none/package.nix:25]:
```suggestion
  installPhase = ''
    runHook preInstall

    # Make it impossible to add to an environment. You should use the appropriate NixOS option.
    # Also leave some breadcrumbs in the file.
    echo "${finalAttrs.name} should not be installed into environments. Please use programs.steam.extraCompatPackages instead." > $out

    install -Dt $steamcompattool compatibilitytool.vdf toolmanifest.vdf
    install -Dt $steamcompattool -m755 launch.sh

    runHook postInstall

  '';
```

File: pkgs/by-name/st/steam-play-none/package.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+stdenvNoCC.mkDerivation (finalAttrs: {
+  name = "Steam-Play-None";
+  version = "0-unstable-2022-12-15";
+  src = fetchFromGitHub {
+    repo = finalAttrs.name;
+    owner = "Scrumplex";
+    rev = "42e38706eb37fdaaedbe9951d59ef44148fcacbf";
+    hash = "sha256-sSHLrB5TlGMKpztTnbh5oIOhcrRd+ke2OUUbiQUqoh0=";
+  };
+  outputs = [ "out" "steamcompattool" ];
```
Comment (Scrumplex) [inline pkgs/by-name/st/steam-play-none/package.nix:14]:
```suggestion
  };
  buildInputs = [ bash ];
  outputs = [ "out" "steamcompattool" ];
```

**Review by Scrumplex — COMMENTED** [review]
I think we should copy (or `install` the files instead of linking, so we can patch the `#!/bin/bash` shebang in `launch.sh`.

**Comment by Scrumplex** [comment]
Result of `nixpkgs-review pr 314116` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>steam-play-none</li>
    <li>steam-play-none.steamcompattool</li>
  </ul>
</details>

**Review by Scrumplex — APPROVED** [review]

File: pkgs/by-name/st/steam-play-none/package.nix [hunk 6]
```diff
@@ -0,0 +1,43 @@
+{ fetchFromGitHub
+, stdenvNoCC
+, lib
+, bash
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  name = "Steam-Play-None";
+  version = "0-unstable-2022-12-15";
+  src = fetchFromGitHub {
+    repo = finalAttrs.name;
```
Comment (wegank) [inline pkgs/by-name/st/steam-play-none/package.nix:10]:
```suggestion
    repo = "Steam-Play-None";
```

File: pkgs/by-name/st/steam-play-none/package.nix [hunk 7]
```diff
@@ -0,0 +1,43 @@
+{ fetchFromGitHub
+, stdenvNoCC
+, lib
+, bash
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  name = "Steam-Play-None";
```
Comment (wegank) [inline pkgs/by-name/st/steam-play-none/package.nix:7]:
```suggestion
  pname = "steam-play-none";
```

File: pkgs/by-name/st/steam-play-none/package.nix [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    # Make it impossible to add to an environment. You should use the appropriate NixOS option.
+    # Also leave some breadcrumbs in the file.
+    echo "${finalAttrs.name} should not be installed into environments. Please use programs.steam.extraCompatPackages instead." > $out
+
+    install -Dt $steamcompattool compatibilitytool.vdf toolmanifest.vdf
+    install -Dt $steamcompattool -m755 launch.sh
+
+    runHook postInstall
+
+  '';
```
Comment (wegank) [inline pkgs/by-name/st/steam-play-none/package.nix:30]:
```suggestion
  '';
```

File: pkgs/by-name/st/steam-play-none/package.nix [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+  meta = {
+    description = ''
```
Comment (wegank) [inline pkgs/by-name/st/steam-play-none/package.nix:33]: Looks too long for `description`, probably put the instructions in `longDescription`?

File: pkgs/by-name/st/steam-play-none/package.nix (hunk 6 above)
Comment (MatthewCroughan) [reply pkgs/by-name/st/steam-play-none/package.nix:10]: It seems it's not case sensitive anyway, so we'll keep finalAttrs.

File: pkgs/by-name/st/steam-play-none/package.nix (hunk 9 above)
Comment (MatthewCroughan) [reply pkgs/by-name/st/steam-play-none/package.nix:32]: I don't agree. I took this directly from https://github.com/NixOS/nixpkgs/blob/29fe096e2030dd8741bb97dc1c06547816ac957f/pkgs/by-name/pr/proton-ge-bin/package.nix#L46-L50

Comment (wegank) [reply pkgs/by-name/st/steam-play-none/package.nix:32]: According to [meta.chapter.md](https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/meta.chapter.md#description-var-meta-description), `description` should be one-line.

I guess I'll leave it as is and if anyone wants to fix that, create a PR that targets both packages.

Comment (MatthewCroughan) [reply pkgs/by-name/st/steam-play-none/package.nix:32]: hmm, then proton-ge-bin should be updated then

File: pkgs/by-name/st/steam-play-none/package.nix (hunk 2 above)
Comment (wegank) [reply pkgs/by-name/st/steam-play-none/package.nix:40]: Let's also keep this for now, I don't think the script has much use on aarch64-linux or darwin.

**Review by wegank — APPROVED** [review]

File: pkgs/by-name/st/steam-play-none/package.nix (hunk 6 above)
Comment (Scrumplex) [reply pkgs/by-name/st/steam-play-none/package.nix]: https://github.com/NixOS/nixpkgs/issues/277994

**Pull request merged**